storage/badger/keys.go

```diff
@@ -115,14 +114,26 @@ func ledgerValueKey(registerID flowgo.RegisterID, blockHeight uint64) []byte {
 // changelog key.
 func registerIDFromLedgerChangelogKey(key []byte) (flowgo.RegisterID, error) {
 	registerString := strings.TrimPrefix(string(key), ledgerChangelogKeyPrefix+"-")
-	parts := strings.SplitN(registerString, "-", 3)
-	if len(parts) < 3 {
+
+	c := strings.Count(registerString, "-")
+	if c == 3 { // old format
+		parts := strings.SplitN(registerString, "-", 3)
+		if len(parts) < 3 {
+			return flowgo.RegisterID{}, fmt.Errorf("failed to parse register ID from %s", string(key))
+		}
+		return flowgo.RegisterID{
+			Owner: parts[0],
+			Key:   parts[2], // skip the controller
+		}, nil
+	}
+	// else is the new format
```

Comment:
Do we need this backwards-compatibility. Can we just remove it?

Reply:
I thought we don't need it initially, but Janez is right people might have use persistant storage that have these values, it seems there is no harm to keep it in case we have to load things.

Reply:
Sounds good!